File: OC/显示层动画/WGOCAnimationPrimary/WGOCAnimationPrimary/位置&形状&颜色&渐变动画/淡入淡出和颜色/WGAlphaAniViewController.m
//
//  WGAlphaAniViewController.m
//  WGOCAnimationPrimary
//
//  Created by wanggang on 2019/5/21.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGAlphaAniViewController.h"

@interface WGAlphaAniViewController ()

@property (nonatomic, strong) UIButton *btn;

@end

@implementation WGAlphaAniViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = UIColor.whiteColor;
    self.title = @"淡入淡出和颜色";
    
    self.btn = [[UIButton alloc] initWithFrame:CGRectMake(20, 100, SCREEN_WIDTH-20*2, 30)];
    [_btn setTitle:@"登录" forState:UIControlStateNormal];
    [_btn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    _btn.backgroundColor = [UIColor cyanColor];
    _btn.alpha = 0;
    [self.view addSubview:_btn];
}

-(void)viewWillAppear:(BOOL)animated{
    
    [super viewWillAppear:animated];
    
    [UIView animateWithDuration:2 animations:^{
        
        self.btn.backgroundColor = UIColor.redColor;
        self.btn.alpha = 1;
    }];
}

@end

File: Swift/显示层初级动画/WGAnimationPrimary/WGAnimationPrimary/综合位置&形状&颜色&渐变动画/淡入淡出和颜色/WGAlphaAniViewController.swift
//
//  WGAlphaAniViewController.swift
//  WGAnimationPrimary
//
//  Created by wanggang on 2019/4/18.
//  Copyright © 2019 bozhong. All rights reserved.
//

import UIKit

class WGAlphaAniViewController: UIViewController {

   
    var loginBtn: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.backgroundColor = UIColor.white
        
        loginBtn = UIButton(frame: CGRect(x: 20, y: 200, width: WgWith-20*2, height: 30))
        loginBtn.backgroundColor = UIColor.cyan
        loginBtn.setTitle("登录", for: .normal)
        loginBtn.setTitleColor(UIColor.black, for: .normal)
        loginBtn.alpha = 0;
        view.addSubview(loginBtn)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        UIView.animate(withDuration: 2) {
            
            self.loginBtn.backgroundColor = UIColor.red
            self.loginBtn.alpha = 1;
        }
    }

}
